Refactor edit pane directive

furinvader opened the pull request:
This is just about cleaning the code a little.
Related to https://community.openproject.org/work_packages/16364/activity, although not changing any functionality.

furinvader committed:
Move EditableFieldsState to inplace-edit directory

furinvader committed:
Remove save callback

furinvader committed:
Move discard function to EditableFieldsState

furinvader committed:
Move WorkPackageService to new components location

furinvader committed:
Refactor edit pane directive scope

Move scope methods that don't
manipulate the DOM to the controller
as it's their place to be.

furinvader committed:
Remove possible event listener memory leak

furinvader committed:
Remove redundant event caller function

furinvader committed:
Remove redundant setFailure function

furinvader committed:
Remove redundant fieldController.state reference

furinvader committed:
Remove unused dependencies

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://furinvader-features-refactoring-16364-edit-pane.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://furinvader-features-refactoring-16364-edit-pane.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

furinvader committed:
Fix url jump query for OpenProject object

If there is no content attribute for the given
meta tag, no parameters are appended. This
prevents legacy tests from failing randomly.

furinvader committed:
Pass correct parameter to pending attachments event

furinvader committed:
Remove references by using the delete keyword

Assuming, using 'null' as reference would have
the same effect as deleting it, was basically
wrong.
